nixos/modules/services/security/oauth2_proxy_nginx.nix

```diff
@@ -45,9 +61,10 @@ in
           proxy_pass_request_body           off;
         '';
       };
-      locations."/".extraConfig = ''
+      "@redirectToAuth2ProxyLogin".return = "307 https://${cfg.domain}/oauth2/start?rd=$scheme://$host$request_uri";
```

Comment:
The naming of the location doesn't seem right. More importantly, does this not mean only https is supported?

Reply:
We need a dummy location, so that the error_page below can be a return to a different domain. For that I we must use a named location because writing return does not work with error_page. See https://nginx.org/en/docs/http/ngx_http_core_module.html#:~:text=%20the%20%E2%80%9C%40%E2%80%9D%20prefix%20defines%20a%20named%20location.%20such%20a%20location%20is%20not%20used%20for%20a%20regular%20request%20processing%2C%20but%20instead%20used%20for%20request%20redirection.%20they%20cannot%20be%20nested%2C%20and%20cannot%20contain%20nested%20locations.%20